solr/core/src/java/org/apache/solr/cloud/ActiveReplicaWatcher.java

```diff
@@ -149,11 +147,6 @@ public synchronized boolean onStateChanged(Set<String> liveNodes, DocCollection
       log.debug("-- already done, exiting...");
       return true;
     }
-    if (collectionState.getZNodeVersion() == lastZkVersion) {
```

Comment:
Do we need this check for non-prs collection?

Reply:
Just want to confirm my understanding, this check did not work for PRS enabled collection because PRS entries change would not update the zNodeVersion, hence PRS updates are ignored and the watch can stall because it never sees the active replica?

If that's the case, i'm wondering if we can check both the node and the child version if `collectionState.isPerReplicaState()` is `true`?

I looked at the original change of this and it's a part of a pretty big PR so I don't know the importance of this check (like did we really see alot of spurious calls?). But erring on the safe side, perhaps we can keep it and make it work for PRS too :) ?

As for reference, I searched message that contains `seconds for replica to become active` (`AddReplica` timeout if this was an issue) on BQ solros nodes, so far we don't have any hits yet.

But regardless, this is definitely an issue for PRS enabled collections and need to be fixed! 😊